src/main/java/seedu/mtracker/console/AddForexParser.java

```diff
@@ -2,11 +2,63 @@
 
 import seedu.mtracker.commands.AddForexCommand;
 import seedu.mtracker.commands.AddInstrumentCommand;
+import seedu.mtracker.ui.TextUi;
 
 public class AddForexParser extends AddInstrumentParser {
+    public static String INSTRUMENT_TYPE = "forex";
+
+    public String getForexRemarksFromUser() {
+        TextUi.displayAddRemarksInstruction();
+        return getUserInput();
+    }
+
+    public String getForexExpiryFromUser() {
+        TextUi.displayAddExpiryInstruction();
+        return getUserInput();
+    }
+
+    public String getForexEntryFromUser() {
+        TextUi.displayAddEntryPriceInstruction();
+        return getUserInput();
+    }
+
+    public String getForexExitFromUser() {
+        TextUi.displayAddExitPriceInstruction();
+        return getUserInput();
+    }
+
+    public void addForexRemarksToParameter() {
+        String remarks = getForexRemarksFromUser();
+        parameters.add(remarks);
+    }
+
+    public void addForexEntryToParameter() {
+        String entryPrice = getForexEntryFromUser();
+        parameters.add(entryPrice);
+    }
+
+    public void addForexExitToParameter() {
+        String exitPrice = getForexExitFromUser();
+        parameters.add(exitPrice);
+    }

```

Comment:
Do include the isValidPrice Conditional to prevent users from inputting a non valid number. Considering that user can choose not to input a price, so need to check if price is empty or if its not empty check if its a validPrice.
Can define a constant number 0 or -1 to indicate that the price is empty.
Feel free to create a separate issue from this and resolve in another PR

Reply:
Yea I will create a separate issue for this and update in new PR.